### ISSUE TITLE
Tensorflow Import on OSX 10.11.5 fails 

### ISSUE THREAD

**Issue by ashvsdata** [tensorflow/tensorflow#3494]
GitHub issues are for bugs / installation problems / feature requests.  
For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### Environment info
Operating System: 10.11.5
Python Version: 2.7.12 (installed from homebrew)

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
 /usr/local/cuda/lib/libcuda.dylib
 /usr/local/cuda/lib/libcudadevrt.a -> /Developer/NVIDIA/CUDA-7.5/lib/libcudadevrt.a
/usr/local/cuda/lib/libcudart.7.5.dylib -> /Developer/NVIDIA/CUDA-7.5/lib/libcudart.7.5.dylib
/usr/local/cuda/lib/libcudart.dylib -> /Developer/NVIDIA/CUDA-7.5/lib/libcudart.dylib
/usr/local/cuda/lib/libcudart_static.a -> /Developer/NVIDIA/CUDA-7.5/lib/libcudart_static.a
/usr/local/cuda/lib/libcudnn.5.dylib
/usr/local/cuda/lib/libcudnn.dylib -> libcudnn.5.dylib
/usr/local/cuda/lib/libcudnn_static.a
If installed from binary pip package, provide:

1. Which pip package you installed.
I tried both install from source using the OSX guide. 
And, the nightly binary with GPU support. 0.90
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.dylib locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.dylib locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.dylib locally
fish: 'python -c "import tensorflow"' terminated by signal SIGSEGV (Address boundary error)

Python quits 

If installed from source, provide 

1. The commit hash (`git rev-parse HEAD`)
5161e4c51b994b3feb93cdb851479c29a3450f31
2. The output of `bazel version`
Build label: 0.3.0-homebrew
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Jul 12 11:11:47 2016 (1468321907)
Build timestamp: 1468321907
Build timestamp as int: 1468321907

### Steps to reproduce
1. Install tensorflow from source or nightly using pip install
2. python -c "import tensorflow"

### What have you tried?
1. import numpy, import scipy etc looking at previous errors. 
2. dtruss output attached for import tensorflow

### Logs or other output that would be helpful
(If logs are large, please upload 
[dtruss_python_tensorflow.txt](https://github.com/tensorflow/tensorflow/files/382028/dtruss_python_tensorflow.txt)
as attachment).

**Comment by caisq** [comment]
Can you try re-installing CUDA with "brew cask install cuda"?

**Comment by ashvsdata** [comment]
Hi @michaelisard  I tried that, but same result. did a bazel clean, and complete rebuild.

**Comment by michaelisard** [comment]
@martinwicke is this behavior you recognize?

**Comment by dmotz** [comment]
I'm seeing the same issue with Python 3.

**Comment by kunaldeo** [comment]
Same issue here on 10.12 with Python 3.5.2:

```
>>> import tensorflow
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.dylib locally
Process 62685 stopped
* thread #1: tid = 0x8c054, 0x00007fffb77c3bf2 libsystem_c.dylib`strlen + 18, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x00007fffb77c3bf2 libsystem_c.dylib`strlen + 18
libsystem_c.dylib`strlen:
->  0x7fffb77c3bf2 <+18>: pcmpeqb (%rdi), %xmm0
    0x7fffb77c3bf6 <+22>: pmovmskb %xmm0, %esi
    0x7fffb77c3bfa <+26>: andq   $0xf, %rcx
    0x7fffb77c3bfe <+30>: orq    $-0x1, %rax

```

**Comment by martinwicke** [comment]
The question is, why does this work on our test machines? @yifeif any idea?

Can you try in a virtualenv? It often provides useful isolation. That said, this seems pervasive.

**Comment by kunaldeo** [comment]
Above result is in virtualenv.

**Comment by jramapuram** [comment]
@alexcolburn 's solution [here](https://github.com/tensorflow/tensorflow/issues/2940) solves this for me

**Comment by ashvsdata** [comment]
Thank you @jramapuram. The solution works!

**Comment by yaroslavvb** [comment]
BTW, the problem that @alexcolbun's solution fixes was is a crash in cuda_diagnostics code which gets triggered when the library can't find cuda dyld. The code tries to print some strings obtained `CFStringGetCStringPtr` which can return null, which shouldn't be printed. I fixed one such condition in #3448 but there were other improper uses in that file I didn't touch. The way to tell if that's the issue is to look at MacOS crash logs (Application->Utilities->Console, under User Diagnostic Reports) and see if there's something like  this in the stack trace

```
perftools::gputools::cuda::Diagnostician::LogDriverVersionInformation() + 512
3   _pywrap_tensorflow.so         	0x00000`

```

**Comment by martinwicke** [comment]
I'll close this as duplicate of #2940.
